oligo_designer_toolsuite/database/_oligos_database.py

```diff
@@ -1,444 +1,465 @@
         self.species = species
-        self.genome_assembly = genome_assembly
         self.annotation_release = annotation_release
-        self.files_source = files_source
-        self.oligo_length_max = oligo_length_max
-        self.oligo_length_min = oligo_length_min
-        self.min_oligos_per_gene = min_oligos_per_gene
+        self.genome_assembly = genome_assembly
+
         if n_jobs is None:
             n_jobs = cpu_count()
         self.n_jobs = n_jobs
 
-        self.file_oligos_DB_tsv = None
-        self.file_oligos_DB_gtf = None
-        self.file_oligos_DB_fasta = None
-        self.oligos_DB = None
-
-        self.sequence_file = sequence_file
-        self.annotation_file = annotation_file
-        # create index file
-        self.transcripts = None
-        pyfaidx.Fasta(self.sequence_file)
-        self.oligos = OligosGenerator(
-            self.oligo_length_min,
-            self.oligo_length_max,
-            self.sequence_file,
-            self.n_jobs,
-        )
+        self.dir_output = os.path.abspath(os.path.join(dir_output, "oligo_database"))
+        Path(self.dir_output).mkdir(parents=True, exist_ok=True)
+
+        # Initialize databse object
+        self.database = {}
         self.oligosets = (
             {}
         )  # will be used later in the gereration of non overlpping sets
 
-    def read_oligos_DB(
+        # Initialize the file for regions with insufficient oligos
+        self.file_removed_regions = os.path.join(
+            dir_output, "regions_with_insufficient_oligos.txt"
+        )
+        with open(self.file_removed_regions, "a") as handle:
+            handle.write(f"Region\tPipeline step\n")
+
+    def create_oligo_database(
         self,
-        format: str,
-        file_oligos_DB_tsv: str = None,
-        file_oligos_DB_gtf: str = None,
-        file_oligos_DB_fasta: str = None,
+        region_ids: list[str] = None,
     ):
         """
-        Create the oligo db dictionary from a file. It can take both a tsv file of a gtf and fasta file and the format of file to process is defined by ``format``.
-
-        :param format: format of file to process
-        :type format: {'tsv', 'gtf'}
-        :param file_oligos_DB_gtf: Path to the file.
-        :type file_oligos_DB_gtf: str
-        :param file_oligos_DB_fasta: Path to the file.
-        :type file_oligos_DB_fasta: str
-        :param file_oligos_DB_tsv: path of the oligos_DB file
-        :type file_oligos_DB_tsv: str
+        Creates the database containing all the oligo sequence extracted form the given sequences (fasta file)
+        and belonging the the specified region. If no list of region IDs is specified then all regions available
+        in the fasta file will be used. The database created is not written automatically to the disk,
+        the ``save_oligo_database`` method has to be called separately.
 
+        :param region_ids: List of regions for which the oligos should be generated, defaults to None
```

Comment:
Can the region ID be different form the gene id? Otherwise it seems to unclear how this list should look like

Reply:
yes, it should just be an id that is used in the fasta header, e.g. if someone uses enhancer regions instead of genes, they could give those regions IDs as well (e.g. enhancer_xyz) and a subset can be given as list of region ids to filter on. It just makes it more generic to other regions (not restricted to genes).